### DISCUSSION TITLE
How to retrieve recordings and transcripts after receiving a notification?

### DISCUSSION THREAD

**Discussion by MikeYeager** [microsoftgraph/msgraph-sdk-dotnet#2486] [Q&A]
I'm using Microsoft.Identity.Web.GraphServiceClient 2.18.1

I was able to successfully create notification subscriptions for recording creation and transcript creation. The notification contains resource property that looks something like this (shortened for brevity):

"resource": "communications/onlineMeetings('MSo5NzZmN ... XhAdGhyZWFkLnYy')/recordings('VjIjIzE5NzZmNG ... JlNzI=')"

and the docs say I can use this to download the recording (or transcript). But it seems this is a portion of a URL. Is there some way to use GraphServiceClient to do the download or do I need to make an HTTPS call using an HTTPClient? The only way I can see to download a recording or transcript with GraphServiceClient is if I have the MeetingId and RecordingId (or TranscriptId), which are not included in the notification.

Thanks in advance, Mike

**Comment by MikeYeager** [comment]
Let me expand on this a little bit. I'm subscribing to tenant wide notifications when transcripts and recordings are created. The notifications look something like this:
```
{
    "value": [{
            "tenantId": "24db396b-b795-45c9-bcfa-d3559193f2f7",
            "changeType": "created",
            "subscriptionId": "cea3e742-6ff8-41cf-b01d-93ca384fbe53",
            "resource": "communications/onlineMeetings('MSo1O ... kLnYy')/transcripts('MSM ... 0ZjU5')",
            "clientState": "AD4777EA-A883-4329-A225-9FFCB8811134",
            "resourceData": {
                "id": "MSMjMCMjODNmMjYxMjEtYTNmMi00ZDgzLWJkN2UtZGQzYTQwYmM0ZjU5",
                "@odata.type": "#Microsoft.Graph.callTranscript",
                "@odata.id": "communications/onlineMeetings('MSo1O ... kLnYy')/transcripts('MSM ... 0ZjU5')"
            },
            "subscriptionExpirationDateTime": "2024-05-16T18:45:42.89747-07:00"
        }
    ]
}
```
Based on this response, is the any way to retrieve the transcript or recording? The only way I can find mentioned in the docs is to have the organizer id, which isn't part of the notification. It does not appear you can download a transcript or recording with only the meeting Id and transcript/recording Id. It seems this notification doesn't have the info necessary.

**Comment by marekrehora** [comment]
Hello Mike, did you ever solve this? I hit a same issue and this is all there is on the internet.

**Comment by MikeYeager** [comment]
@marekrehora Unfortunately, no. Next step is to make a connection with the team inside Microsoft. I'm getting nowhere with this or other public discussions. It's not yet critical for us as we can get around it for testing purposes by designating an account and making that a co-organizer on the meetings we test with, but it will become critical soon.

**Comment by MikeYeager** [comment]
I'm just coming back to this and there doesn't seem to have been any movement. So far, I've been unlucky finding a connection within Microsoft. I'm hoping that this comment causes the team to take another look and perhaps respond.

**Comment by JamesDerrickHH** [comment]
is it not the case that the resource includes the meetingId, and the transcriptId, so you can extract them and use them in other graph calls to get the actual transcripts?
I've just started my journey with this, currently not getting any data from what the subscription is sending, but it was my assumption that that'd be the next step

**Comment by MikeYeager** [comment]
@JamesDerrickHH @marekrehora  My fault for not closing this sooner. At some point, a few months back, MSFT changed the payload to include the information I was missing and retrieving the video started working. We ended up using Whisper to create a more accurate transcript from the video than Teams provided, so did not finish retrieving the transcript, though we will likely go back and do that so we can use it to determine who is speaking as our transcript doesn't provide that.
